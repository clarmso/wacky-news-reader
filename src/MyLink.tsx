import { Link } from "@material-ui/core";

type MyLinkProps = {
  href: string;
  isInherit?: boolean;
};

const MyLink: React.FC<MyLinkProps> = (props) => {
  return (
    <Link
      target="_blank"
      rel="noreferrer"
      color={props.isInherit ? "inherit" : "primary"}
      href={props.href}
    >
      {props.children}
    </Link>
  );
};

export default MyLink;
