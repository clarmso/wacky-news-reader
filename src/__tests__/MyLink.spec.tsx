import { render } from "@testing-library/react";
import MyLink from "../MyLink";

it("renders properly", () => {
  const { asFragment } = render(
    <MyLink href="https://duckduckgo.com">Duck!</MyLink>
  );
  const renderedHtml = asFragment();
  expect(renderedHtml).toMatchSnapshot();
});

it("set isInherit property", () => {
  const { asFragment } = render(
    <MyLink isInherit={true} href="https://duckduckgo.com">
      Duck!
    </MyLink>
  );
  const renderedHtml = asFragment();
  expect(renderedHtml).toMatchSnapshot();
});
