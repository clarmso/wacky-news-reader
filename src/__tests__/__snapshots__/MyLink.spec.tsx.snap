// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<DocumentFragment>
  <a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
    href="https://duckduckgo.com"
    rel="noreferrer"
    target="_blank"
  >
    Duck!
  </a>
</DocumentFragment>
`;

exports[`set isInherit property 1`] = `
<DocumentFragment>
  <a
    class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
    href="https://duckduckgo.com"
    rel="noreferrer"
    target="_blank"
  >
    Duck!
  </a>
</DocumentFragment>
`;
